using-vertx/src/main/java/org/acme/vertx/PgClientProducer.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.acme.vertx;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import io.reactiverse.axle.pgclient.PgClient;
+import io.reactiverse.axle.pgclient.PgPool;
+import io.reactiverse.pgclient.PgPoolOptions;
+import io.vertx.axle.core.Vertx;
+
+@ApplicationScoped
+public class PgClientProducer {
+
+    @Inject
+    @ConfigProperty(name = "pg.pool.database", defaultValue = "quarkus_test")
+    String database;
+
+    @Inject
+    @ConfigProperty(name = "pg.pool.user", defaultValue = "quarkus_test")
+    String user;
+
+    @Inject
+    @ConfigProperty(name = "pg.pool.password", defaultValue = "quarkus_test")
+    String password;
+
+    @Inject
+    @ConfigProperty(name = "pg.schema.create", defaultValue = "true")
+    boolean schemaCreate;
+
+    @Inject
+    Vertx vertx;
+
+    @Produces
+    PgPool client;
+
+    @PostConstruct
+    void config() {
+        PgPoolOptions options = new PgPoolOptions()
```

Comment:
I would have prefered an extension producing the client from the configuration instead of letting the user creating the instance.

Reply:
This would be ideal. Can you file an issue and assign to me? I can work on it later and update the guide.